core/src/main/java/com/microsoft/applicationinsights/internal/heartbeat/HeartBeatProvider.java

```diff
@@ -0,0 +1,304 @@
+package com.microsoft.applicationinsights.internal.heartbeat;
+
+import com.microsoft.applicationinsights.TelemetryClient;
+import com.microsoft.applicationinsights.TelemetryConfiguration;
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+import com.microsoft.applicationinsights.internal.shutdown.Stoppable;
+import com.microsoft.applicationinsights.internal.util.ThreadPoolUtils;
+import com.microsoft.applicationinsights.telemetry.MetricTelemetry;
+import com.microsoft.applicationinsights.telemetry.Telemetry;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+/**
+ * <p>
+ *  Concrete implementation of Heartbeat functionality. This class implements
+ *  {@link com.microsoft.applicationinsights.internal.heartbeat.HeartBeatProviderInterface} and
+ *  {@link com.microsoft.applicationinsights.internal.shutdown.Stoppable}
+ * </p>
+ *
+ * @author Dhaval Doshi
+ * @since 03-30-2018
+ */
+public class HeartBeatProvider implements HeartBeatProviderInterface, Stoppable {
+
+  /**
+   * The name of the heartbeat metric.
+   */
+  private final String HEARTBEAT_SYNTHETIC_METRIC_NAME = "HeartbeatState";
+
+  /**
+   * The list of disabled properties
+   */
+  private List<String> disableDefaultProperties = new ArrayList<>();
+
+  /**
+   * List of disabled heartbeat providers
+   */
+  private List<String> disabledHeartBeatPropertiesProviders = new ArrayList<>();
+
+  /**
+   * The counter for heartbeat sent to portal
+   */
+  private AtomicLong heartbeatsSent;
+
+  /**
+   * Map to hold heartbeat properties
+   */
+  private ConcurrentMap<String, HeartBeatPropertyPayload> heartbeatProperties;
+
+  /**
+   * Interval at which heartbeat would be sent
+   */
+  private long interval;
+
+  /**
+   * Telemetry client instance used to send heartbeat.
+   */
+  private TelemetryClient telemetryClient;
+
+  /**
+   * ThreadPool used for adding properties to concurrent dictionary
+   */
+  private ExecutorService propertyUpdateService;
```

Comment:
Since this is single use, I don't think it's needed as a field. Local variable in constructor should be fine.